src/back/glsl/mod.rs

```diff
@@ -2428,23 +2430,30 @@ impl<'a, W: Write> Writer<'a, W> {
             Expression::Unary { op, expr } => {
                 use crate::{ScalarKind as Sk, UnaryOperator as Uo};
 
-                write!(
-                    self.out,
-                    "({} ",
-                    match op {
-                        Uo::Negate => "-",
-                        Uo::Not => match *ctx.info[expr].ty.inner_with(&self.module.types) {
-                            TypeInner::Scalar { kind: Sk::Sint, .. } => "~",
-                            TypeInner::Scalar { kind: Sk::Uint, .. } => "~",
-                            TypeInner::Scalar { kind: Sk::Bool, .. } => "!",
-                            ref other =>
-                                return Err(Error::Custom(format!(
-                                    "Cannot apply not to type {:?}",
-                                    other
-                                ))),
-                        },
+                let ty = ctx.info[expr].ty.inner_with(&self.module.types);
+
+                match *ty {
+                    TypeInner::Vector { kind: Sk::Bool, .. } => {
+                        write!(self.out, "not(")?;
```

Comment:
What about the other operations like `Negate` and `Not` for non booleans?

Reply:
They are handled below.